tests/units/Toolbox.php

```diff
@@ -874,8 +874,26 @@ public function testGetMailServerProtocols(
       $type = \Toolbox::parseMailServerConnectString($cnx_string)['type'];
 
       $this->string($type)->isEqualTo($expected_type);
-      $this->variable(\Toolbox::getMailServerProtocolClassname($type))->isEqualTo($expected_protocol);
-      $this->variable(\Toolbox::getMailServerStorageClassname($type))->isEqualTo($expected_storage);
+
+      if ($expected_protocol !== null) {
+         $this->object(\Toolbox::getMailServerProtocolInstance($type))->isInstanceOf($expected_protocol);
+      } else {
+         $this->variable(\Toolbox::getMailServerProtocolInstance($type))->isNull();
+      }
+
+      /*
+       * I do not know how to test it.
+       * I had this error: "ErrorException: fsockopen(): unable to connect to localhost:143".
```

Comment:
dovecot is only started form `imap` directory tests. I'm pretty sure you will not be able to have any mock working as expected with a static call somewhere.

Reply:
I moved these tests in imap test suite and it works now.